src/JsonSchema/BackwardCompatibleSchemaFactory.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\JsonSchema;
+
+use ApiPlatform\Metadata\Operation;
+
+/**
+ * This factory decorates range integer and number properties to keep Draft 4 backward compatibility.
+ *
+ * @see https://github.com/api-platform/core/issues/6041
+ *
+ * @internal
+ */
+class BackwardCompatibleSchemaFactory implements SchemaFactoryInterface, SchemaFactoryAwareInterface
+{
+    public function __construct(private readonly SchemaFactoryInterface $decorated)
+    {
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function buildSchema(string $className, string $format = 'json', string $type = Schema::TYPE_OUTPUT, Operation $operation = null, Schema $schema = null, array $serializerContext = null, bool $forceCollection = false): Schema
+    {
+        $schema = $this->decorated->buildSchema($className, $format, $type, $operation, $schema, $serializerContext, $forceCollection);
+
+        foreach ($schema->getDefinitions() as $definition) {
+            foreach ($definition['properties'] as $property) {
+                /** @var \ArrayObject $property */
+                if (\in_array($property['type'], ['integer', 'number'], true)) {
+                    if ($property->offsetExists('exclusiveMinimum')) {
```

Comment:
why not `isset($property['exclusiveMinimum')`?

Reply:
Honestly, no reason. I will change it to `isset` 👍